src/pfaffian.jl

```diff
@@ -0,0 +1,58 @@
+using LinearAlgebra: exactdiv
+
+# in-place O(n³) algorithm to compute the exact Pfaffian of
+# a skew-symmetric matrix over integers (or potentially any ring supporting exact division).
+#
+#     G. Galbiati & F. Maffioli, "On the computation of pfaffians,"
+#     Discrete Appl. Math. 51, 269–275 (1994).
+#     https://doi.org/10.1016/0166-218X(92)00034-J
+function exactpfaffian!(A::AbstractMatrix{<:Integer})
```

Comment:
I would just rename `exactpfaffian!` to `pfaffian!` etcetera.

I would also split it into two functions in order to support `SkewHermitian` matrices more efficiently:
```jl
function _pfaffian!(A::AbstractMatrix{<:Integer})
    ... same as above, but with no argument checks ...
end

function pfaffian!(A::AbstractMatrix{<:Integer})
    LinearAlgebra.require_one_based_indexing(A)
    isskewhermitian(A) || throw(ArgumentError("Pfaffian requires a skew-Hermitian matrix"))
    return _pfaffian!(A)
end
pfaffian!(A::SkewHermitian{<:Integer}) = _pfaffian!(A.data)
```
Note also that you will want an additional method somewhere for `pfaffian(A::SkewHermTridiagonal{<:Real})` that just multiplies every other off-diagonal element.